### PR TITLE
Update e2e tests to handle snapshot csi tests

### PR DESCRIPTION
**Pull request by nbalacha**
This commit provides a way to configure the e2e tests to run csi snapshot
tests only for thin device classes.

Signed-off-by: N Balachandran <nibalach@redhat.com>

### CODE REVIEW

**Comment by nbalacha** [comment]
@llamerada-jp , this is the initial set of changes for enabling the csi sanity snapshot tests for thin deviceclasses. Please let me know if this is what you had in mind. 
I can then update the workflow like this:
`--- a/.github/workflows/e2e-k8s-workflow.yaml
+++ b/.github/workflows/e2e-k8s-workflow.yaml
@@ -16,10 +16,12 @@ jobs:
       fail-fast: false
       matrix:
         kubernetes_versions: ["1.24.2", "1.23.6", "1.22.9"]
+        snapshot_tests: ["true", "false"]
     env:
       KUBERNETES_VERSION: ${{ matrix.kubernetes_versions }}
       TEST_SCHEDULER_MANIFEST: ${{ inputs.test_scheduler_manifest }}
       SKIP_NODE_FINALIZE: ${{ inputs.skip_node_finalize }}
+      TEST_THIN_DEVICECLASS: ${{ matrix.snapshot_tests }}
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
`

File: e2e/sanity_test.go [hunk 1]
```diff
@@ -62,5 +65,13 @@ func testSanity() {
 		_, _, err = kubectl("exec", "-n", "topolvm-system", "daemonset/topolvm-node", "--", "test", "-e", path)
 		return sanity.PathIsOther, err
 	}
+
+	if os.Getenv("TEST_THIN_DEVICECLASS") == "true" {
+		// csi.storage.k8s.io/fstype=xfs,topolvm.cybozu.com/device-class=sbvglvm
```
Comment (peng225) [inline e2e/sanity_test.go:70]: What is `sbvglvm`? (The device set is actually set to `thin` at line 73.)

File: e2e/sanity_test.go [hunk 2]
```diff
@@ -62,5 +65,13 @@ func testSanity() {
...
+		volParams := make(map[string]string)
+		volParams["csi.storage.k8s.io/fstype"] = "xfs"
+		volParams["topolvm.cybozu.com/device-class"] = "thin"
+		tc.TestVolumeParameters = volParams
```
Comment (peng225) [inline e2e/sanity_test.go:74]: I'm afraid I could not find any documents about `TestVolumeParameters`.
I would be happy if you could share the document you referred to when you write that code.

**Comment by llamerada-jp** [comment]
@nbalacha 
The way I requested it before was incorrect. Would you please add a new test parameter like below?
the test of snapshot will be skipped in default (equal to the current patern) and will be executed in the new case.

https://github.com/topolvm/topolvm/blob/a00f86f7183260982988af771c487b8678822b0f/.github/workflows/e2e-k8s.yaml#L14-L26

https://github.com/topolvm/topolvm/blob/a00f86f7183260982988af771c487b8678822b0f/.github/workflows/e2e-k8s-workflow.yaml#L2-L9

File: e2e/sanity_test.go (hunk 2 above)
Comment (nbalacha) [reply e2e/sanity_test.go:74]: I found it in the csi-test code:

https://github.com/kubernetes-csi/csi-test/blob/862e81af7f4b61e965714f8c528ee8ca91780dd7/pkg/sanity/sanity.go#L53

This describes the various parameters that can be specified.

File: e2e/sanity_test.go (hunk 1 above)
Comment (nbalacha) [reply e2e/sanity_test.go:70]: I'll fix this.

**Comment by nbalacha** [comment]
> @nbalacha The way I requested it before was incorrect. Would you please add a new test parameter like below? the test of snapshot will be skipped in default (equal to the current patern) and will be executed in the new case.
> 
> https://github.com/topolvm/topolvm/blob/a00f86f7183260982988af771c487b8678822b0f/.github/workflows/e2e-k8s.yaml#L14-L26
> 
> https://github.com/topolvm/topolvm/blob/a00f86f7183260982988af771c487b8678822b0f/.github/workflows/e2e-k8s-workflow.yaml#L2-L9

Thank you @llamerada-jp . I will make the changes.

File: e2e/sanity_test.go (hunk 1 above)
Comment (nbalacha) [reply e2e/sanity_test.go]: Fixed.

File: .github/workflows/e2e-k8s-workflow.yaml [hunk 3]
```diff
@@ -16,10 +19,12 @@ jobs:
       fail-fast: false
       matrix:
         kubernetes_versions: ["1.24.2", "1.23.6", "1.22.9"]
+        snapshot_tests: ["true", "false"]
```
Comment (peng225) [inline .github/workflows/e2e-k8s-workflow.yaml:22]: This parameter is not used anywhere.

Comment (nbalacha) [reply .github/workflows/e2e-k8s-workflow.yaml]: Thank you. Fixed.

**Review by peng225 — APPROVED** [review]

**Review by llamerada-jp — APPROVED** [review]

**Pull request merged**